src/collective/taxonomy/widget.py

```diff
@@ -1,17 +1,48 @@
+from time import time
+from plone.memoize import ram
 import zope.interface
 import zope.component
 import zope.schema.interfaces
 
 from z3c.form import interfaces
 from z3c.form.browser.orderedselect import OrderedSelectWidget
 from z3c.form.widget import FieldWidget
+from z3c.form.widget import SequenceWidget
+from z3c.form.browser import widget
 
 from interfaces import ITaxonomySelectWidget
 
+ONE_DAY = 60 * 60 * 24
 
+
+def _cache_one_day(fun, self):
+    key = '{0}{1}'.format(
+        self.field.__name__,
+        time() // ONE_DAY
+    )
+    return key
+
```

Comment:
Can you add another component to the key (not sure what) so that the cache is invalidated if you change the taxonomy ttw or by import?